frame/staking/src/pallet/impls.rs

```diff
@@ -1088,8 +1062,8 @@ impl<T: Config> ElectionDataProvider for Pallet<T> {
 		<Validators<T>>::remove_all();
 		#[allow(deprecated)]
 		<Nominators<T>>::remove_all();
-
-		T::VoterList::unsafe_clear();
+		// TODO: sort it out some other way
```

Comment:
I would get rid of the ReadOnly abstraction, it does not solve any important problems, it just creates some 🤷

Reply:
That being said, it seems like this is available for benchmarks despite being "read-only"

Reply:
Yeah, since it's an `unsafe` operation and it's available only for benches - I deemed it reasonable to leave it there for simplicity.

As for the ReadOnly abstraction - it does solve a problem of separating concerns and being able to pass ReadOnly lists to whoever does not need the whole implementation, without having to implement it.

The alternative is we'd have to implement a whole interface per list for StakeTracker, for example, which would render the write methods as noop. 
Yeah, it's a bit clunky due to Rust's inheritance specifics, but ultimately better than having to implement the interface every time we want to restrict access? What do you say about that?